src/sentry/remote_config/docs/api.md

```diff
@@ -44,16 +68,30 @@ Retrieve the DSN's configuration.
 
 ### Set Configuration [POST]
 
-Set the DSN's configuration.
+Set the DSN's configuration. A hash value of `null` is provided if the object does not exist. If the API finds an existing configuration object an error is returned. If the hash value is provided then it will be compared against the hash of the remote value. A new hash is always returned on successful write. If the hashes do not match an error is returned.
```

Comment:
Since DSNs can be disabled and the app replaced with a new DSN, for the same project. Should we tie this to "project config" instead?

```suggestion
Set the project's configuration. A hash value of `null` is provided if the object does not exist. If the API finds an existing configuration object an error is returned. If the hash value is provided then it will be compared against the hash of the remote value. A new hash is always returned on successful write. If the hashes do not match an error is returned.
```

Here is an example of 2 DSNs for a single project:

![image](https://github.com/getsentry/sentry/assets/1633368/94208fb1-c32d-4e32-ac3a-9447b9704abc)

Reply:
The point is to split the configurations by DSN.  DSNs allow people to split configurations up.  I.e. the next.js case where client and server share a project.

If a DSN needs to be rotated we can expose UI to copy the configuration data over.

Reply:
I'm not sure this is ideal. Specially now that we're going to have 1 DSN for muliple apps. So a new field for to split config furthers will be used (like `application`).

What do we gain by scoping this to DSN instead of just the project?

Reply:
What I mentioned above is what we gain.  If Sentry wants to introduce a new scoping concept then we can follow that migration.  I'm tying to the DSN because it exists today.

Reply:
It's fine if we want to scope to the project.  I'm just stating the reasoning for scoping to the DSN.  If that's not wanted we can drop it.

Reply:
Decided offline.  Will scope to the project for now.  DSN is imperfect for scoping.